dashboards-observability/public/components/visualizations/charts/stats/stats.tsx

```diff
@@ -0,0 +1,317 @@
+  const selectedOrientation = dataConfig?.chartStyles?.orientation || Orientation;
+  let orientation = selectedOrientation === 'auto' || selectedOrientation === 'v' ? 'auto' : 'h';
+  const selectedTextMode = dataConfig?.chartStyles?.textMode || StatsTextMode;
+  const textMode =
+    selectedTextMode === 'auto' || selectedTextMode === 'values+names' ? 'auto' : selectedTextMode;
+  const textColor = dataConfig?.chartStyles?.textColor?.childColor || TextColor;
+  const textAlign = dataConfig?.chartStyles?.textAlign || TextAlignment;
+
+  let autoChartLayout = {
+    xaxis: {
+      visible: false,
+      showgrid: false,
+      anchor: 'y1',
+      margin: {
+        l: 0,
+        r: 0,
+        b: 0,
+        t: 0,
+      },
+    },
+    yaxis: {
+      visible: false,
+      showgrid: false,
+      anchor: 'x1',
+      margin: {
+        l: 0,
+        r: 0,
+        b: 0,
+        t: 0,
+      },
+    },
+  };
+
+  const [statsData, statsLayout]: Plotly.Data[] = useMemo(() => {
+    let calculatedStatsData: Plotly.Data[] = [];
+    if (dimensionsLength || metricsLength) {
+      // case 1,2: no dimension, single/multiple metrics for chart type horizontal/text
+      // will be handled with horizontal/textmode chart
+      // if (!dimensionsLength && metricsLength >= 1) {}
+
+      // case 3: multiple dimensions and multiple metrics for chart type auto
+      if (dimensionsLength && metricsLength) {
+        const selectedDimensionsData = dimensions
+          .map((dimension: any) => data[dimension.name])
+          .reduce((prev, cur) => {
+            return prev.map((i, j) => `${i},<br>${cur[j]}`);
+          });
+        calculatedStatsData = metrics.map((metric: any, metricIndex: number) => {
+          autoChartLayout = {
+            ...autoChartLayout,
+            [`yaxis${metricIndex > 0 ? metricIndex + 1 : ''}`]: {
+              visible: false,
+              showgrid: false,
+              anchor: `x${metricIndex > 0 ? metricIndex + 1 : ''}`,
+              margin: {
+                l: 0,
+                r: 0,
+                b: 0,
+                t: 0,
+              },
+            },
+            [`xaxis${metricIndex > 0 ? metricIndex + 1 : ''}`]: {
+              visible: false,
+              showgrid: false,
+              anchor: `y${metricIndex > 0 ? metricIndex + 1 : ''}`,
+              margin: {
+                l: 0,
+                r: 0,
+                b: 0,
+                t: 0,
+              },
+            },
+          };
+
+          return {
+            x: selectedDimensionsData,
+            y: data[metric.label],
+            fill: 'tozeroy',
+            mode: 'line',
+            type: 'scattergl',
+            name: metric.label,
+            ...(metricIndex > 0 && {
+              xaxis: `x${metricIndex + 1}`,
+              yaxis: `y${metricIndex + 1}`,
+            }),
+          };
+        });
+
+        // label/text
+        let textLabelTrace: any = {
```

Comment:
I didn't see textLabelTrace is actually changed later we can use const here.

Reply:
updated in next commit